Instruments/source/CacheTupleProducer.cxx

```diff
@@ -0,0 +1,164 @@
+/*! Produce synchronization tree.
+This file is part of https://github.com/hh-italian-group/h-tautau. */
+
+#include "AnalysisTools/Run/include/program_main.h"
+#include "AnalysisTools/Core/include/RootExt.h"
+#include "AnalysisTools/Core/include/TextIO.h"
+#include "h-tautau/Analysis/include/EventInfo.h"
+#include "h-tautau/Core/include/CacheTuple.h"
+#include "h-tautau/Core/include/EventTuple.h"
+#include "h-tautau/Analysis/include/SignalObjectSelector.h"
+
+struct Arguments {
+    REQ_ARG(std::string, input_file);
+    REQ_ARG(std::string, output_file);
+    REQ_ARG(std::string, channels);
+    REQ_ARG(std::string, period);
+    REQ_ARG(std::string, selections);
+    REQ_ARG(std::string, unc_sources);
+    REQ_ARG(std::string, jet_orderings);
+    REQ_ARG(bool, isData);
+    REQ_ARG(bool, hasBjetPair);
+    OPT_ARG(bool, runSVFit, true);
+    OPT_ARG(bool, runKinFit, true);
+    OPT_ARG(Long64_t, maxEvents, std::numeric_limits<Long64_t>::max());
+};
+
+namespace analysis {
+
+class CacheTupleProducer {
+public:
+    using CacheEvent = cache_tuple::CacheEvent;
+    using CacheTuple = cache_tuple::CacheTuple;
+
+    CacheTupleProducer(const Arguments& _args) : args(_args), run_period(Parse<analysis::Period>(args.period()))
+    {
+        EventCandidate::InitializeJecUncertainty(run_period);
+        std::vector<std::string> selections = SplitValueList(args.selections(),false,",");
```

Comment:
It is quite a common operation and it would be useful also for the future if you implement a template version SplitValueListT in AnalysisTools.